### PR TITLE
Use the same language data source as in Page Properties -> Advanced

### LINKED ISSUE #55: Language drop-down should expose both language and optionally country name
The data source for available languages just leverages the `TranslationConfig.getLanguages()` to retrieve the available languages. That returns a Map with key being the combination of country code (optional) with language code (mandatory) (separated by "_") and as value just a label for the language (disregarding the country).

The drop-down being used in Page Properties -> Advanced -> Language (`/libs/wcm/foundation/components/basicpage/v1/basicpage/tabs/advanced/items/column/items/section1/items/language/datasource`) leveraging the data source `cq/gui/components/common/datasources/languages` (with its source being `/libs/cq/gui/components/common/datasources/languages/languages.jsp`) however exposes the following format:

`<Language Label> (<Country Label>)` (without exposing any ISO codes). 

I think that is better suited for choosing the proper language country combination. Particularly when it comes to 'artificial countries' (i.e. markets which span more than one country, like middle africa). Those usually leverage one of the reserved country codes from <https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2#User-assigned_code_elements>. Maybe one can even rather filter on top of the data source `cq/gui/components/common/datasources/languages` in https://github.com/orbinson/aem-dictionary-translator/blob/main/core/src/main/java/be/orbinson/aem/dictionarytranslator/servlets/datasource/AvailableLanguageDatasource.java. That way both drop-downs look the same.

### PR DESCRIPTION
**Pull request by kwin**
Just filter out the existing languages

This closes #55

### CODE REVIEW

**Comment by bdhoine** [comment]
Thanks for the contribution. Let's get this reviewed and merged and see if we can create a new release with all the pending changes.

File: core/src/main/java/be/orbinson/aem/dictionarytranslator/servlets/datasource/LanguageDatasource.java [hunk 1]
```diff
@@ -1,12 +1,17 @@
 package be.orbinson.aem.dictionarytranslator.servlets.datasource;
 
 import be.orbinson.aem.dictionarytranslator.services.DictionaryService;
+import com.adobe.agl.text.Collator;
```
Comment (kwin) [inline core/src/main/java/be/orbinson/aem/dictionarytranslator/servlets/datasource/LanguageDatasource.java:4]: This is the wrong collator

**Review by bdhoine — COMMENTED** [review]
looks good at first sight, we might still need to add some tests to make sure no regressions will occur, I'm also working on some basic automated smoke tests with a frontend framework but I guess the functionality in this one might be better suited for a test with AemMocks

@kwin can you also mention this change in the CHANGELOG.md? 

@royteeuwen can you also do a review?

**Comment by kwin** [comment]
> @kwin can you also mention this change in the CHANGELOG.md?

Done in https://github.com/orbinson/aem-dictionary-translator/pull/56/commits/26a87ba2a94a0f150d84d5a97cf81a39b0209332

File: core/src/main/java/be/orbinson/aem/dictionarytranslator/servlets/datasource/LanguageDatasource.java [hunk 2]
```diff
@@ -19,67 +24,186 @@
...
+                .map(e -> {
+                    if (emitTextFieldResources) {
+                        return TextFieldResource.create(request.getLocale(), request.getResourceResolver(), e.getKey(), e.getValue());
+                    } else {
+                        return ValueTextResource.create(request.getLocale(), request.getResourceResolver(), e.getKey(), e.getValue());
+                    }
+                })
+                .collect(Collectors.toList());
+        // sort by display names
+        Collections.sort(resourceList);
```
Comment (kwin) [inline core/src/main/java/be/orbinson/aem/dictionarytranslator/servlets/datasource/LanguageDatasource.java:138]: which sort order should be used:
a) by language/country name (label)
b) by language/country code

The former was and is used for "Create Language", "Delete Language", and "Create Key". The latter is used for "Edit" (https://github.com/orbinson/aem-dictionary-translator/pull/52). We should consolidate, but which one to pick?

Comment (royteeuwen) [reply core/src/main/java/be/orbinson/aem/dictionarytranslator/servlets/datasource/LanguageDatasource.java:138]: For me it's fine to use the same order as all the other orders in AEM, which one would that be?

File: core/src/main/java/be/orbinson/aem/dictionarytranslator/servlets/datasource/LanguageDatasource.java [hunk 3]
```diff
@@ -19,67 +24,186 @@
...
+     * @param dictionaryPath
+     * @return a map of language/country codes and their display names
+     * @throws IOException 
+     * @throws ServletException 
+     */
+    public static Map<String, String> getAllAvailableLanguages(SlingHttpServletRequest request, @NotNull SlingHttpServletResponse response) throws ServletException, IOException {
+        // TranslationConfig.getLanguages(ResourceResolver) does never return the country label, 
+        // therefore use the data source which is also used in the Page Properties dialog (Advanced Tab in Language)
+        RequestDispatcherOptions options = new RequestDispatcherOptions();
+        options.setForceResourceType("cq/gui/components/common/datasources/languages");
```
Comment (royteeuwen) [inline core/src/main/java/be/orbinson/aem/dictionarytranslator/servlets/datasource/LanguageDatasource.java:86]: nice "fix". Sad that Adobe didn't put this in an actual class ;) why provide a TranslationConfig and not use it

Comment (kwin) [reply core/src/main/java/be/orbinson/aem/dictionarytranslator/servlets/datasource/LanguageDatasource.java:85]: Indeed, poor API.

**Pull request merged**